custom-plugins/cms131v14_diabetes_eye_exam/protocols/cms131v14_protocol.py

```diff
@@ -0,0 +1,690 @@
+            return False
+
+        return True
+
+    def _in_denominator(self, patient: Patient) -> bool:
+        if self._has_hospice_care_in_period(patient):
+            log.info(f"CMS131v14: Patient {patient.id} in hospice care")
+            return False
+
+        if self._is_age_66_plus_with_frailty(
+            patient
+        ) and self._has_advanced_illness_or_dementia_meds(patient):
+            log.info(f"CMS131v14: Patient {patient.id} is age 66+ with frailty and advanced illness or dementia meds")
+            return False
+
+        if self._is_age_66_plus_in_nursing_home(patient):
+            log.info(f"CMS131v14: Patient {patient.id} is age 66+ in nursing home")
+            return False
+
+        if self._has_palliative_care_in_period(patient):
+            log.info(f"CMS131v14: Patient {patient.id} has palliative care")
+            return False
+
+        if self._has_bilateral_absence_of_eyes(patient):
+            log.info(f"CMS131v14: Patient {patient.id} has bilateral eye absence")
+            return False
+
+        return True
+
+    def _in_numerator(self, patient: Patient) -> bool:
+        if self._has_retinopathy_diagnosis_in_period(patient):
+            return self._has_retinal_exam_by_eye_care_professional_in_period(patient)
+
+        if self._has_retinal_exam_in_period_or_year_prior(patient):
+            return True
+
+        if self._has_autonomous_eye_exam_in_period(patient):
+            return True
+
+        if self._has_retinal_finding_with_severity_in_period(patient):
+            return True
+
+        if self._has_retinal_finding_no_severity_in_prior_year(patient):
+            return True
+
+        return False
+
+    def _has_diabetes_diagnosis(self, patient: Patient) -> bool:
+        try:
+            diabetes_conditions = Condition.objects.for_patient(patient.id).active()
+
+            log.info(
+                f"CMS131v14: Found {diabetes_conditions.count()} active conditions for patient {patient.id}"
+            )
+
+            for condition in diabetes_conditions:
+                for coding in condition.codings.all():
+                    if self._is_diabetes_code(coding):
+                        log.info(f"CMS131v14: Found diabetes diagnosis with code {coding.code}")
+                        return True
+
+            log.info(f"CMS131v14: No active diabetes diagnoses found for patient {patient.id}")
+            return False
+
+        except Exception as e:
+            log.error(f"CMS131v14: Error checking diabetes diagnosis: {str(e)}")
+            return False
+
+    def _check_age_18_to_75(self, patient: Patient) -> bool:
+        try:
+            if not getattr(patient, "birth_date", None):
+                log.info("CMS131v14: Missing birth_date; patient fails age check")
+                return False
+
+            birth_date = arrow.get(patient.birth_date)
+            age_years = (self.now - birth_date).days // 365
```

Comment:
Should use `patient.age_at(self.now)`, but also this logic was defined in this file as `def _calculate_age`. The SDK defines a way to determine patient age, but this plugin defines two more.